asyncpg/pool.pyi

```diff
@@ -0,0 +1,355 @@
+import asyncio
```

Comment:
Is there a reason to have a stub for `pool.py`? I'd prefer to have type hints inlined right into the code even if it makes the file longer. I think it will be easier to keep typing in sync with code changes if they are close to each other.

Reply:
The issue here is that `PoolConnectionProxy` is using a metaclass to wrap `Connection` methods, but type hinting does not work well with highly dynamic classes such as this. In cases such as this, you must use a `.pyi` file.

Reply:
So the issue is only about the `PoolConnectionProxy` class, right? Probably, we could move the `PoolConnectionProxy` into a separate file and annotate it using a `.pyi` file. The rest of the `pool.py` could be annotated using usual inline type hints. What do you think, will it work?

Reply:
Yes, I believe that should work.

Reply:
I went ahead and implemented this suggestion